pkg/scheduler/objects/application.go

```diff
@@ -844,6 +850,27 @@ func (sa *Application) tryAllocate(headRoom *resources.Resource, nodeIterator fu
 			}
 			continue
 		}
+
+		requiredNode := request.GetRequiredNode()
+		// does request (daemon set pods?) has any constraint to run on specific node?
+		if requiredNode != "" {
+			node := getnode(requiredNode)
+			alloc := sa.tryNode(node, request)
```

Comment:
What about calling `node.IsSchedulable()` here? Wouldn't that make sense?

Reply:
No, We shouldn't call the method as it is not required for daemon set pod and this `if` block is meant for daemon set pod only. Hence the removal of `isValidFor` method. There could be a situation wherein running a type of daemon set pods itself is mandatory to make the node as "schedulable/ready".